omcproxy: update default configuration to fix startup

tiagogaspar8 opened the pull request:
Omcproxy uses bridges and phisical interfaces instead of the default
configuration file's interface.

If you keep the default omcproxy config it won't start.

ynezz commented:
@tiagogaspar8 could you please kindly read https://openwrt.org/submitting-patches one more time and adjust your contribution accordingly, namely the commit subject and Signed-off-by line?

tiagogaspar8 commented:
OK, I'm sorry, new to this and learning.
I cant change the commit name, should I make a new pull request with the new naming and etc.?

tiagogaspar8 commented:
Ok I managed to do it.
If there are any errors just say the word and I'll fix them.

tiagogaspar8 commented:
Can I be the reviewer?
And on another note is omcproxy still maintained? cause there are issues in the repository that aren't getting fixed.

adrianschmutzler commented:
@tiagogaspar8 You can add a "review" either by stating in a comment that you thoroughly reviewed and/or tested the patch, or you can use GitHub's features to add an Approval to the patch (upper right corner when you enter the "Commits" or "File changes" section of the PR).

I cannot comment on the state of the package.

tiagogaspar8 reviewed (commented):
This patch works and It's been tested for more than a month.

tiagogaspar8 commented:
Well, I can't approve since I was the one that created this patch, but I added a comment saying that this is more than tested. I hope this gets merged

CodeFetch commented:
@tiagogaspar8 Can you please elaborate the issue you experience? I don't understand why removing the "IPv6 configuration section" fixes it. It sounds more like a bug which you circumvent by removing the broken functionality.

tiagogaspar8 commented:
I see what you mean, and from the logs it doesn't seem to disable any kind of ipv6 functionality.
I'll explain my thought process.
In OpenWRT the `wan` and `wan6` are in fact aliases to a real interface that actually has the IP addresses configured, in a standard configuration it is `wan`.
So, with the default configuration, when omcproxy boots, it looks for an interface behind the alias interface, for the alias `lan` the real interface is `br-lan` and for the interfaces for `wan` and `wan6` the respective real interface is `wan`. But since it has both configured, it's going to add `wan` twice.

Another important detail to note is that when you manage to get omcproxy running, the pocess shows like this:
`/usr/sbin/omcproxy wan br-lan scope=global`.

As you can see it has made the "translation" from the alias interface to the real interface and starts the deamon.

Now that you mention that this might be a bug I think there are two options:

1. This is a bug and omcproxy isn't handling more than two interfaces to proxy (not likely)
2. This is not a bug, and omcproxy refuses to have two instances running on the same two interfaces, so it doesn't start.

I believe it's the second option since the deamon uses the real IP interface.

Also, I believe that the package luci should be updated because since it shows both `wan` and `wan6` it may lead to confusion that both interfaces need to be added for it to support IPv4 and 6, and that the script should be updated to spit out more errors instead of leaving the user in the dark.

CodeFetch reviewed (approved):
@tiagogaspar8 Thank you for your clarification. Your change is correct. As the fw3 script queries the zone of the network, only the IPv4 wan network should be added as the IPv4 wan network has the auto setting in OpenWrt by default.

CodeFetch commented:
Ah wait... This breaks the firewall rules...

tiagogaspar8 commented:
That's correct, I'll do it.

tiagogaspar8 committed:
omcproxy: update default configuration to fix startup

There is currently a bug that prevents omcproxy from starting with the
system
Omcproxy uses bridges and physical interfaces instead of the default
configuration file's interface

Signed-off-by: Tiago Gaspar <tiagogaspar8@gmail.com>

tiagogaspar8 commented:
Done.

tiagogaspar8 commented:
Hi everyone,

This has been stale for some time...
@stintel could you give it a friendly push?😁
If needed I can rebase it.

tiagogaspar8 commented:
@sbyx I noticed you're the omcproxy maintainer, could you review this?
